module/zfs/zpl_ctldir.c

```diff
@@ -96,20 +96,47 @@ zpl_root_readdir(struct file *filp, void *dirent, filldir_t filldir)
 #endif /* HAVE_VFS_ITERATE */
 
 /*
+ * Used by zpl*getattr* functions below.
+ */
+static int
+zpl_simple_getattr(const struct path *path, struct kstat *stat,
```

Comment:
I think we should get rid of this function and just call generic_fillattr. That'll make things much easier.
The only difference is simple_getattr `stat->blocks = inode->i_mapping->nrpages << (PAGE_SHIFT - 9);` which we don't really use since the caller are all directories.